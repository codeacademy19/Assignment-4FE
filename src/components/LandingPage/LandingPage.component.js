/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import React, { Component } from 'react';
import {
  Platform, StyleSheet, Text, View, SafeAreaView, ScrollView, Header, Image, TouchableOpacity, TouchableHighlight,
} from 'react-native';
import ActionButton from 'react-native-action-button';
import FormCard from '../FormCard/FormCard.component';
import styles from './LandingPage.css';
import getForms from '../../helpers/getForms';

export default class App extends Component {
    state={
      Forms: [],
    }

    componentDidMount() {
      return getForms('http://localhost:7000/Form').then(async (res) => {
        this.setState({
          Forms: res,
        });
      });
    }

    onImageClick=() => {
      this.componentDidMount();
    }

    onFromClick=(name) => {
      this.props.navigation.navigate('Fill', { name });
    }

    render() {
      const { navigation } = this.props;
      return (
        <SafeAreaView style={styles.container}>
          <View style={styles.header}>
            <Text style={styles.headerText}>Awesome Forms</Text>
            <TouchableOpacity onPress={() => this.onImageClick()}>
              <Image
                className="image"
                style={styles.menu}
                source={require('../../../Icon/menu.png')}
              />
            </TouchableOpacity>
          </View>
          <ScrollView
            contentContainerStyle={{
              flexGrow: 1,
              justifyContent: 'center',
              alignItems: 'center',
              alignSelf: 'center',
            }}
            style={{ width: '200%', backgroundColor: '#cddcf7' }}
          >
            {this.state.Forms.map(Form => (
              <TouchableHighlight onPress={() => this.onFromClick(Form.FormName)} style={styles.formCard}>
                <FormCard name={Form.FormName} time={Form.createdAt} onPress={() => this.onFromClick(Form.FormName)} />
              </TouchableHighlight>
            ))
            }

          </ScrollView>
          <ActionButton
            buttonColor="#1d61e0"
            style={styles.actionButtonIcon}
            onPress={() => this.props.navigation.navigate('New')}
          />

        </SafeAreaView>
      );
    }
}
