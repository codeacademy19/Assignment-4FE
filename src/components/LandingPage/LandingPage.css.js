import { StyleSheet } from 'react-native';
import { FONT_HEADER } from '../../constants';

export default StyleSheet.create({
  scroll: {
    width: '100%',


  },
  actionButtonIcon: {
    fontSize: 20,
    color: 'white',
  },
  formCard: {
    flex: 0.3,
    width: '100%',
    height: 260,
    alignSelf: 'center',
  },
  header: {
    backgroundColor: '#1d61e0',
    width: '100%',
    height: '10%',
    flexDirection: 'row',
    justifyContent: 'center',

  },
  headerText: {
    fontSize: FONT_HEADER,
    color: 'white',
    paddingLeft: 5,
    alignSelf: 'center',
    alignItems: 'flex-start',

  },
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    height: '100%',
    width: '100%',
  },
  menu: {
    alignSelf: 'center',
    alignItems: 'flex-end',
    marginLeft: 100,
    marginTop: 10,
  },
});
