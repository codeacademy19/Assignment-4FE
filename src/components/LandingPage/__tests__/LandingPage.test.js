import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import LandingPage from '../LandingPage.component';

describe('CardView component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <LandingPage />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should call "componentDidMount" on rendering', () => {
    mockFn = jest.fn();
    const wrapper = shallow(<LandingPage />);
    wrapper.instance().componentDidMount = mockFn();
    expect(mockFn).toHaveBeenCalled();
  });
});
