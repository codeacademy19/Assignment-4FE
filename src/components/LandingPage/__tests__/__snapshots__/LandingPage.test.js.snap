// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1d61e0",
        "flexDirection": "row",
        "height": "10%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "color": "white",
          "fontSize": 22,
          "paddingLeft": 5,
        }
      }
    >
      Awesome Forms
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        className="image"
        source={
          Object {
            "testUri": "../../../Icon/menu.png",
          }
        }
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "marginLeft": 100,
            "marginTop": 10,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    style={
      Object {
        "backgroundColor": "#cddcf7",
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        Object {
          "color": "white",
          "fontSize": 20,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            Object {
              "elevation": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.35,
              "shadowRadius": 3,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(29, 97, 224, 1)",
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "height": 56,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 56,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 24,
                    "marginTop": -4,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
